packages/nuxt/src/core/nitro.ts

```diff
@@ -115,6 +115,11 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
       tsConfig: {
         include: [
           join(nuxt.options.buildDir, 'types/nitro-nuxt.d.ts')
+        ],
+        exclude: [
+          ...nuxt.options.modulesDir.map(m => relativeWithDot(nuxt.options.buildDir, m)),
```

Comment:
Note: Consider `modulesDir` can technically be normal directories with parent/sibiling `node_modules` dir) as well. We might have filtered out the ones not ending with `node_modules` here for greater safety.

Reply:
That's a nice idea. I've just copied across our current default in the nuxt tsconfig so if we filter it further we should update them in parallel.